### PR TITLE
frontend changes for Nextjs Framework

### PR DESCRIPTION
**Pull request by bhokaremoin**

### CODE REVIEW

File: gui/src/components/LayoutComponents/SideBar.tsx [hunk 1]
```diff
@@ -24,34 +35,55 @@ const SideBar: React.FC = () => {
...
+              handleSelectedOption(option)
+                ? 'sidebar_item_selected'
+                : 'sidebar_item'
+            } flex flex-col items-center justify-center gap-1 text-white`}
+            onClick={() => handleOptionClick(option)}
+          >
+            <Image
+              width={24}
+              height={24}
+              sizes={'100vw'}
```
Comment (Parth2506) [inline gui/src/components/LayoutComponents/SideBar.tsx:70]: sizes is not necessary if you are passing proper values for width and height

File: gui/src/components/LayoutComponents/SideBar.tsx [hunk 2]
```diff
@@ -24,34 +35,55 @@ const SideBar: React.FC = () => {
...
 
+  const filteredOptions = useMemo(() => {
+    const projectType = getProjectTypeFromFramework(projectFramework);
+    if (projectType === projectTypes.DESIGN) {
+      return sidebarOptions.filter((option) =>
+        ['design', 'workbench'].includes(option.id),
+      );
+    } else if (projectType === projectTypes.BACKEND) {
+      return sidebarOptions.filter((option) =>
+        ['board', 'workbench', 'code', 'pull_request'].includes(option.id),
```
Comment (Parth2506) [inline gui/src/components/LayoutComponents/SideBar.tsx:46]: not a good idea to have all the names here, create a constant  and below it an constant array with all the items and use it here ensuring in future more can be added without issue

File: gui/src/components/LayoutComponents/SideBar.tsx [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
 
 const SideBar: React.FC = () => {
   const router = useRouter();
   const pathname = usePathname();
+  const [projectFramework, setProjectFramework] = useState<string | null>(null);
+
+  useEffect(() => {
+    let storedProjectType = null;
+    if (typeof window !== undefined) {
+      storedProjectType = localStorage.getItem('projectFramework');
```
Comment (Parth2506) [inline gui/src/components/LayoutComponents/SideBar.tsx:19]: storedProjectType is redundant here

File: gui/src/components/DesignStoryComponents/story.module.css [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    /* To Do Tag */
+    background: #474747;
+}
+
+.purple_tag {
+    /* In Progress Tag */
+    background: #a05bd5;
+}
+
+.green_tag {
```
Comment (Parth2506) [inline gui/src/components/DesignStoryComponents/story.module.css:64]: move the common ones to _app.css

Comment (Parth2506) [reply gui/src/components/DesignStoryComponents/story.module.css:64]: like purple_tag, green_tag, and so on

File: gui/src/components/DesignStoryComponents/ReviewList.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+              return (
+                <div
+                  key={'review_item_' + index}
+                  className={`${styles.review_item} flex flex-row gap-4 px-2 py-3`}
+                  onClick={() => openReview(story.id)}
+                >
+                  <div
+                    className={`${styles.review_image_container} relative flex h-[60px] w-[60px] items-center justify-center overflow-hidden rounded-lg`}
+                  >
+                    <img
```
Comment (Parth2506) [inline gui/src/components/DesignStoryComponents/ReviewList.tsx:48]: don't use  tag

File: gui/src/components/DesignStoryComponents/DesignStoryList.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    storyList,
+    setSelectedStory,
+  } = useDesignContext();
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleOpenStoryDetails = (storyDetails) => {
+    setOpenStoryDetailsModal(true);
+    setSelectedStory(storyDetails);
+    localStorage.setItem('storyId', storyDetails.id);
+  };
+  const tagCSS = {
```
Comment (Parth2506) [inline gui/src/components/DesignStoryComponents/DesignStoryList.tsx:22]: we have similar const for board as well, so can we have a single const and import it from one of the constant files

**Review by Parth2506 — CHANGES_REQUESTED** [review]

File: gui/src/components/SyntaxDisplay/SyntaxDisplay.tsx [hunk 7]
```diff
@@ -4,15 +4,18 @@ import { dracula } from 'react-syntax-highlighter/dist/esm/styles/prism';
...
 }
 
-const SyntaxDisplay: React.FC<SyntaxDisplayProps> = ({ error }) => {
+const SyntaxDisplay: React.FC<SyntaxDisplayProps> = ({ type, msg }) => {
   return (
     <>
-      <span className={'text-sm font-normal'}>
-        AI Developer has an error in the terminal.
-      </span>
+      {type === 'ERROR' && (
```
[the remark this answers is not in the document]
Comment (moinContlo) [reply gui/src/components/SyntaxDisplay/SyntaxDisplay.tsx:14]: ok making a constant named ActivityLogType

File: gui/src/components/LayoutComponents/SideBar.tsx [hunk 8]
```diff
@@ -24,34 +35,55 @@ const SideBar: React.FC = () => {
...
+            id={`${option.id}_${index}_option`}
+            className={`${
+              handleSelectedOption(option)
+                ? 'sidebar_item_selected'
+                : 'sidebar_item'
+            } flex flex-col items-center justify-center gap-1 text-white`}
+            onClick={() => handleOptionClick(option)}
+          >
+            <Image
+              width={24}
```
[the remark this answers is not in the document]
Comment (moinContlo) [reply gui/src/components/LayoutComponents/SideBar.tsx:68]: ok understood

File: gui/src/components/LayoutComponents/SideBar.tsx (hunk 2 above)
Comment (moinContlo) [reply gui/src/components/LayoutComponents/SideBar.tsx:46]: ok

File: gui/src/components/DesignStoryComponents/DesignStoryList.tsx (hunk 6 above)
Comment (moinContlo) [reply gui/src/components/DesignStoryComponents/DesignStoryList.tsx]: resolved

File: gui/src/components/DesignStoryComponents/ReviewList.tsx (hunk 5 above)
Comment (moinContlo) [reply gui/src/components/DesignStoryComponents/ReviewList.tsx]: resolved

File: gui/src/components/DesignStoryComponents/story.module.css (hunk 4 above)
Comment (moinContlo) [reply gui/src/components/DesignStoryComponents/story.module.css]: resolved

File: gui/src/components/LayoutComponents/SideBar.tsx (hunk 3 above)
Comment (moinContlo) [reply gui/src/components/LayoutComponents/SideBar.tsx]: removed

File: gui/src/components/LayoutComponents/SideBar.tsx (hunk 1 above)
Comment (moinContlo) [reply gui/src/components/LayoutComponents/SideBar.tsx]: got it, resolved

File: gui/src/components/LayoutComponents/SideBar.tsx [hunk 9]
```diff
@@ -24,34 +35,55 @@ const SideBar: React.FC = () => {
...
+            key={index}
+            id={`${option.id}_${index}_option`}
+            className={`${
+              handleSelectedOption(option)
+                ? 'sidebar_item_selected'
+                : 'sidebar_item'
+            } flex flex-col items-center justify-center gap-1 text-white`}
+            onClick={() => handleOptionClick(option)}
+          >
+            <Image
```
[the remark this answers is not in the document]
Comment (moinContlo) [reply gui/src/components/LayoutComponents/SideBar.tsx]: resolved

**Pull request closed without merging**